### ISSUE TITLE
Add CodeStream comment doesn't work for Slack.

### ISSUE THREAD

**Issue by Shaddy** [TeamCodeStream/CodeStream#20]
Using Slack integration: I just select any file and any line of code, but the message is not sent. It shows red buttons as if something has failed.

![image](https://user-images.githubusercontent.com/247221/49078854-6b307780-f23f-11e8-9c80-451d46675824.png)

This is not working in one of my VsCode instances, I've it installed on another computer and it works as expected. 

Any suggestion to see whats happening?

**Comment by planteater** [comment]
@Shaddy - Sorry to hear that you're having issues. Just to confirm, you signed in with the same Slack credentials/workspace on both computers? I would suggestion signing out and then re-authenticating with Slack just to make sure we've got all of our scopes correct.

If you're still having issues, some log files would be hepfull. Here are the details:

1. In your VS Code user settings (see gear menu at bottom left of VSC), search for CodeStream and change the Trace Level setting to "Debug".
2. Reload VS Code by going to the VS Code command palette (Cmd + Shift + P) and then searching for Reload Window.
3. Try posting again, and if it still fails open the VS Code Output view with Cmd + Shift + U and select "CodeStream" from the dropdown at the top-right of that pane.
4. Copy everything there and paste it into a text file to send to me (dave@codestream.com).
5. Change the dropdown selection to "CodeStream (Agent)" and grab that output as well.

Thanks!

**Comment by Shaddy** [comment]
> Sorry to hear that you're having issues. Just to confirm, you signed in with the same Slack credentials/workspace on both computers? I would suggestion signing out and then re-authenticating with Slack just to make sure we've got all of our scopes correct

This is the first I tried without success. I just created a log as you pointed out, expect an email from me.

Thanks for your time!

**Comment by akonwi** [comment]
@Shaddy after logging in again, have you tried to post a message with code again?

**Comment by Shaddy** [comment]
Yes, and unfortunately it's not working :(

**Comment by eamodio** [comment]
@Shaddy Sorry for the issues but can you try again with CodeStream v0.30? We think it will help.

**Comment by Shaddy** [comment]
Hi guys, 

I have no problem, just want to help solve this.

Using **0.30.1** with just an update and no resign.  The problem is still there. Doing log-out/sign-in the problem is also there. The new interface for posting code/issues/trap/bookmark is really **amazing**!!

I sent logs for the last version to dave, do you want me to record with the new version?

**Comment by planteater** [comment]
New logs would be great. Thanks!

**Comment by Shaddy** [comment]
I isolated the log part that happens when trying to send again the failing snippet code.

```
[2018-12-07 13:01:50:283] API(04e68269-2ab7-4e38-9ce1-6086f03e684e): FAILED(0x) GET /streams/undefined • 393 ms
[Error - 2:01:50 PM] [2018-12-07 13:01:50:283] [274] CodeStreamApiProvider.getStream — failed • 393 ms
Error: Forbidden(RAPI-1003): Object not found
Error: Forbidden(RAPI-1003): Object not found
    at c:\Users\Sherab\.vscode\extensions\codestream.codestream-0.30.1\dist\agent.js:186:81257
    at slackApi_SlackApiProvider.createPost (c:\Users\Sherab\.vscode\extensions\codestream.codestream-0.30.1\dist\agent.js:186:81865)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

At the beginning is also failing.

```
[Error - 2:01:49 PM] [2018-12-07 13:01:49:830] [263] SlackApiProvider.createPost — failed • 407 ms
TypeError: Cannot read property '0' of undefined
TypeError: Cannot read property '0' of undefined
    at c:\Users\Sherab\.vscode\extensions\codestream.codestream-0.30.1\dist\agent.js:186:81257
    at slackApi_SlackApiProvider.createPost (c:\Users\Sherab\.vscode\extensions\codestream.codestream-0.30.1\dist\agent.js:186:81865)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

Let me know if this is enough to throw a clue or you need the entire log.

**Comment by Shaddy** [comment]
Can't attach because it seems that contains an `accessToken`, not sure if its what I think it is. But can't be public. I can send it to any *@codestream.com email tho.

**Comment by marcelofarias** [comment]
Please send it to marcelo at codestream

**Comment by Shaddy** [comment]
Finally, as Marcelo pointed out if the code you're sharing doesn't have a remote configured it will fail. Don't know if this is a correct behavior but in case someone has the same problem, **add a remote to your git repo**.

**Comment by planteater** [comment]
Fixed in 0.30.2 - CodeStream is much more valuable when working with repos that have remotes, but commenting on code from a repo without a remote should no longer result in an error.
